parcels/util/_helpers.py

```diff
@@ -0,0 +1,53 @@
+"""Internal helpers for Parcels."""
+
+import functools
+import warnings
+from typing import Callable
+
+PACKAGE = "Parcels"
+
+
+def deprecated(msg: str = "") -> Callable:
+    """Decorator marking a function as being deprecated
+
+    Parameters
+    ----------
+    msg : str, optional
+        Custom message to append to the deprecation warning.
+
+    Examples
+    --------
+    ```
+    @deprecated("Please use `another_function` instead")
+    def some_old_function(x, y):
+        return x + y
+
+    @deprecated()
+    def some_other_old_function(x, y):
+        return x + y
+    ```
+    """
+    if msg:
+        msg = " " + msg
```

Comment:
Should this not be an f-string?

Reply:
I think this is one of the first cases I've had where I found `+` to be more readable (just reading like english; "adding a space to the front"). Happy to update if you disagree.

Reply:
OK, is fine then